apps/api-service/src/transformer.ts

```diff
@@ -1,3 +1,90 @@
+import type { SuperJSONResult } from "superjson";
 import { SuperJSON } from "superjson";
 
-export const transformer = SuperJSON;
+/**
+ * Creates a transformer that wraps SuperJSON and warns about unsupported data types.
+ * Specifically, it detects class instances that haven't been registered with SuperJSON,
+ * as these cannot be properly serialized and may cause issues.
+ */
+function createWarningTransformer() {
+  const warnedTypes = new Set<string>();
```

Comment:
this set should not be required if you change to an assert approach like i mentioned in my other comment